src/ORM/PolymorphicHasManyList.php

```diff
@@ -89,8 +74,9 @@ public function getForeignRelationKey(): string
      * @param string $foreignField The name of the composite foreign (has_one) relation field. Used
      * to generate the ID, Class, and Relation foreign keys.
      * @param class-string<TForeign> $foreignClass Name of the class filter this relation is filtered against
+     * @param string $foreignRelation The name of the has_many relation this list is filtered by, if any.
      */
-    public function __construct($dataClass, $foreignField, $foreignClass)
+    public function __construct(string $dataClass, string $foreignField, string $foreignClass, ?string $foreignRelation = null)
```

Comment:
Why is a new param being added here?

Reply:
Because that was always the intention. As per the deprecation notice:
> @deprecated 5.2.0 Will be replaced with a parameter in the constructor